acceptance/tests/parser_functions/puppet_lookup_cmd.rb

```diff
@@ -6,12 +6,11 @@
 
 ### @testroot = "/etc/puppetlabs"
 @testroot = master.tmpdir("#{@module_name}")
-
 @coderoot = "#{@testroot}/code"
 @confdir = "#{@testroot}/puppet"
-
-@mastername = on(master, facter('fqdn')).stdout.chomp
-@agentname = on(agent, facter('fqdn')).stdout.chomp
+@agent = [agent].flatten.find { |host| host != master }
```

Comment:
Agreed, seems strange this test assumes a single agent. Most of them are written to apply iteratively over agents.